### PR TITLE
Added NestedComment model and all relations

### PR DESCRIPTION
**Pull request by jamescavallo**
Relations, routes and NestedComments. Fixed spec and Heroku

### CODE REVIEW

**Comment by itsjoyce** [comment]
looks good!

File: bin/rails [hunk 1]
```diff
@@ -2,8 +2,3 @@
 APP_PATH = File.expand_path("../config/application", __dir__)
 require_relative "../config/boot"
 require "rails/commands"
-if ENV['RAILS_ENV'] == 'test'
```
Comment (jpaints3) [inline bin/rails:5]: why is this deleted? does that remove simplecov?

File: db/schema.rb [hunk 2]
```diff
@@ -21,29 +29,44 @@
     t.datetime "updated_at", null: false
   end
 
+  create_table "nested_comments", force: :cascade do |t|
+    t.string "text"
```
Comment (jpaints3) [inline db/schema.rb:33]: I don't think this needs the text. Nested Comments are just like the postTags relation.

**Review by jpaints3 — APPROVED** [review]
I have a question about some deletions and the nested comment relation seems to have too much. Aside for that lgtm

File: bin/rails (hunk 1 above)
Comment (jamescavallo) [reply bin/rails:5]: This was the cause of double simplecov

File: db/schema.rb (hunk 2 above)
Comment (jamescavallo) [reply db/schema.rb:33]: Why wouldn't it need text?  Its a comment

**Review by itsjoyce — APPROVED** [review]

File: bin/rails (hunk 1 above)
Comment (jpaints3) [reply bin/rails:5]: ok sounds good
